### DISCUSSION TITLE
[과목 1] 제 1장 데이터 모델링의 이해

### DISCUSSION THREAD

**Discussion by white-gyu** [flataex/sqlp-study#2] [Custom 연습문제] [answered]
**문제 1. 주어진 데이터에서 아래의 SQL문이 수행된 결과로 옳은 것은?**

|col1|col2|
|:---:|:---:|
|100|100|
|null|60|
|null|null|

```sql
select coalesce(col1, col2 * 50, 50)
from sqld_39_12;
```

1. 100, 3000, 50
2. 100, null, 50
3. 100, 60, 50
4. 100, 3000, null

**문제 2. 다음 계층형 쿼리문에 대한 설명으로 옳지 않은 것은?**

|id|parent_id|name|parent_name|depth|
|:---:|:---:|:---:|:---:|:---:|
|3|0|A||1|
|4|0|B||1|
|5|3|C|A|2|
|6|3|D|A|2|
|7|3|E|A|2|
|8|3|F|A|2|
|9|6|G|F|3|
|10|4|H|B|2|
|11|4|I|B|2|

```sql
select id, parent_id, name, parent_name
from sqld39_21
where parent_id not in (3)
start with parent_id = 0
connect by prior id = parent_id
order siblings by parent_id asc, id asc;
```

1. PARENT_ID가 0이라도 3이 포함되면 전개를 멈춘다.
2. 순방향 전개다.
3. 중복이 생겼을 때 루프를 돌지 않기 위해 NO CYCLE 옵션을 사용할 수 있다.
4. ORDER SIBLINGS BY를 하면 전체 테이블 기준으로 정렬한다.

### 참고 자료

[SQLD 39회 기출 문제 ( 50문제 / 정답,해설포함 )](https://yunamom.tistory.com/265)

**Comment by beolggoolosori** [comment, marked as answer]
> 문제 1

**1.100, 3000, 50**

1행: coalesce(100, 100 * 50, 50) → 첫 번째 값이 NULL이 아니므로 100 반환
2행: coalesce(NULL, 60 * 50, 50) → 첫 번째 값이 NULL이므로 두 번째 값 3000 반환
3행: coalesce(NULL, NULL * 50, 50) → 첫 번째 값과 두 번째 값이 모두 NULL이므로 세 번째 값 50 반환

> 문제 2

**4. ORDER SIBLINGS BY는 계층 구조에서 같은 부모를 가진 노드끼리 정렬된다.**

id | parent_id | name | parent_name
-- | -- | -- | --
3 | 0 | A |  
4 | 0 | B |  
10 | 4 | H | B
11 | 4 | I | B
